fbw-a32nx/src/systems/systems-host/systems/FWC/PseudoFWC.ts

```diff
@@ -1332,6 +1503,194 @@ export class PseudoFWC {
+    );
+
+    const discPbPressedAfterDisconnection =
+      !this.autoPilotDisengagedInstantPulse.read() &&
+      (this.autoPilotInstinctiveDiscPressedPulse.read() || masterWarningButtonLeft || masterWarningButtonRight);
+
+    this.autoPilotOffUnacknowledged.write(
+      this.autoPilotDisengagedInstantPulse.read(),
+      apEngaged || this.autoPilotInstinctiveDiscPressedTwiceInLast1p9Sec.read() || discPbPressedAfterDisconnection,
+    );
+
+    this.autoPilotOffInvoluntaryMemory.write(
+      !apEngaged && !this.autoPilotOffVoluntaryMemory.read() && this.autoPilotOffUnacknowledged.read(),
+      !this.autoPilotOffUnacknowledged.read(),
+    );
+    this.autoPilotOffInvoluntary.set(this.autoPilotOffInvoluntaryMemory.read());
+    this.autoPilotOffShowMemo.set(this.autoPilotOffVoluntaryMemory.read() || this.autoPilotOffInvoluntaryMemory.read());
+
+    if (this.autoPilotDisengagedInstantPulse.read()) {
+      // Request quiet CRC one time
+      this.requestMasterWarningFromApOff = true;
+      this.soundManager.setVolume(FwsAuralVolume.Attenuated);
+      this.soundManager.enqueueSound('cavalryChargeOnce'); // On the A320, play first cav charge completely no matter what
+    }
+    if (!this.autoPilotOffVoluntaryFirstCavalryChargeActive.read()) {
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.setVolume(FwsAuralVolume.Full);
+    }
+    if (!this.autoPilotOffVoluntaryMemory.read() && !this.autoPilotOffInvoluntaryMemory.read()) {
+      this.requestMasterWarningFromApOff = false;
+      this.soundManager.dequeueSound('cavalryChargeOnce');
+      this.soundManager.dequeueSound('cavalryChargeCont');
+      this.soundManager.setVolume(FwsAuralVolume.Full);
+    }
+
+    this.autoPilotInstinctiveDiscPressedPulse.write(false, deltaTime);
+
+    // approach capability downgrade. Debounce first, then suppress for a certain amount of time
+    // (to avoid multiple triple clicks, and a delay which is too long)
+    let fmgcApproachCapability = 0;
+    const getApproachCapability = (dw4: Arinc429WordData): number => {
+      let appCap = 0;
+      const landModeActive = dw4.bitValueOr(14, false);
+      const land2Capacity = dw4.bitValueOr(23, false);
+      const land3FailPassiveCapacity = dw4.bitValueOr(24, false);
+      const land3FailOperationalCapacity = dw4.bitValueOr(25, false);
+
+      if (land2Capacity) {
+        appCap = 2;
+      } else if (land3FailPassiveCapacity) {
+        appCap = 3;
+      } else if (land3FailOperationalCapacity) {
+        appCap = 4;
+      } else if (landModeActive) {
+        appCap = 1;
+      } else {
+        appCap = 0;
+      }
+      return appCap;
+    };
+
+    if (this.fmgc1DiscreteWord4.get().isNormalOperation()) {
+      fmgcApproachCapability = getApproachCapability(this.fmgc1DiscreteWord4.get());
+    } else if (this.fmgc2DiscreteWord4.get().isNormalOperation()) {
+      fmgcApproachCapability = getApproachCapability(this.fmgc1DiscreteWord4.get());
+    }
```

Comment:
```suggestion
    if (this.fmgc1DiscreteWord4.get().isNormalOperation() || this.fmgc2DiscreteWord4.get().isNormalOperation()) {
      fmgcApproachCapability = getApproachCapability(this.fmgc1DiscreteWord4.get());
    }
```

Reply:
That's a typo in the original code, it's meant to take either Fmgc1 or Fmgc2 discreteword4. Latest commit fixes this